telebot/async_telebot.py

```diff
@@ -4014,6 +4023,10 @@ async def send_document(
         if reply_parameters and (reply_parameters.allow_sending_without_reply is None):
             reply_parameters.allow_sending_without_reply = self.allow_sending_without_reply
 
+        if isinstance(document, types.InputFile) and visible_file_name:
+            # inputfile name ignored, warn
```

Comment:
`files[key] = (value.file_name, value.file)`

visible_file_name will be ignoread, isn't it?